src/libraries/System.Data.OleDb/src/System/Data/ProviderBase/DbBuffer.cs

```diff
@@ -354,11 +348,17 @@ protected override bool ReleaseHandle()
             base.handle = IntPtr.Zero;
             if (IntPtr.Zero != ptr)
             {
-                Interop.Kernel32.LocalFree(ptr);
+                LocalFree(ptr);
             }
             return true;
         }
 
+        private static unsafe nint LocalAllocZeroed(int initialSize) =>
+            (nint)Interop.Kernel32.LocalAllocZeroed((uint)initialSize);
+
+        private static unsafe void LocalFree(nint ptr) =>
+            Interop.Kernel32.LocalFree((void*)ptr);
```

Comment:
The more appropriate thing to do here would be to update all places that are using `nint` as a pointer value with `void*`. Wrapping just to change types is undesirable for these APIs.

Reply:
Yes, agreed, consumers should ideally use `void*`.

Reply:
Removed the helper methods.